encodings/runend/src/compress.rs

```diff
@@ -20,19 +21,21 @@ pub fn runend_encode(array: &PrimitiveArray) -> (PrimitiveArray, PrimitiveArray)
         let (ends, values) = runend_encode_primitive(array.maybe_null_slice::<$P>());
 
         let mut compressed_values = PrimitiveArray::from_vec(values, validity);
-        compressed_values.statistics().set(Stat::IsConstant, false.into());
         compressed_values.statistics().set(Stat::RunCount, compressed_values.len().into());
         array.statistics().get(Stat::Min).map(|s| compressed_values.statistics().set(Stat::Min, s));
         array.statistics().get(Stat::Max).map(|s| compressed_values.statistics().set(Stat::Max, s));
+        array.statistics().get(Stat::IsConstant).map(|s| compressed_values.statistics().set(Stat::IsConstant, s));
         array.statistics().get(Stat::IsSorted).map(|s| compressed_values.statistics().set(Stat::IsSorted, s));
         array.statistics().get(Stat::IsStrictSorted).map(|s| compressed_values.statistics().set(Stat::IsStrictSorted, s));
 
         let compressed_ends = PrimitiveArray::from(ends);
+        compressed_ends.statistics().set(Stat::IsConstant, (compressed_ends.len() > 1).into());
         compressed_ends.statistics().set(Stat::IsSorted, true.into());
         compressed_ends.statistics().set(Stat::IsStrictSorted, true.into());
-        compressed_ends.statistics().set(Stat::IsConstant, false.into());
-        compressed_ends.statistics().set(Stat::Max, array.len().into());
-        compressed_ends.statistics().set(Stat::RunCount, compressed_ends.len().into());
+        if (compressed_ends.len() > 0) {
```

Comment:
Freshly created ends array will have Min of 0 and Max of the array.len(). You don't need to get scalars